config/config.go

```diff
@@ -57,3 +58,161 @@ func FindAndReadConfig(fs afero.Fs, configFile string) (*v1.Config, error) {
 	c, err2 := ReadConfig(reader)
 	return c, err2
 }
+
+type ver struct {
+	Version int `json:"version"`
+}
+
+func detectVersion(b []byte) (int, error) {
+	v := &ver{}
+	err := json.Unmarshal(b, v)
+	if err != nil {
```

Comment:
You can probably get away with not having this branch

Reply:
Because if err !=nil then version == 0?

Reply:
can return the `calculatedVersion,err` even if there's an error

Reply:
Ah yes, I was worried about `v` being nil, but that can't be the case.